src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/SwitchHelper.groovy

```diff
@@ -726,4 +727,74 @@ class SwitchHelper {
     static SwitchPropertiesDto getCachedSwProps(SwitchId switchId) {
         return getCachedAllSwProps().find { it.switchId == switchId }
     }
+
+    static def setServer42FlowRttForSwitch(Switch sw, boolean isServer42FlowRttEnabled) {
+        def originalProps = northbound.get().getSwitchProperties(sw.dpId)
+        if (originalProps.server42FlowRtt != isServer42FlowRttEnabled) {
+            def s42Config = sw.prop
+            northbound.get().updateSwitchProperties(sw.dpId, originalProps.jacksonCopy().tap {
+                server42FlowRtt = isServer42FlowRttEnabled
+                server42MacAddress = s42Config ? s42Config.server42MacAddress : null
+                server42Port = s42Config ? s42Config.server42Port : null
+                server42Vlan = s42Config ? s42Config.server42Vlan : null
+            })
+        }
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            def actualRuleCookies = s42SwitchRules(sw.dpId)
+            //one rule per vlan/vxlan
+            int expectedNumberOfS42Rules = isVxlanEnabled(sw.dpId) ? 2 : 1
+
+            if (isServer42FlowRttEnabled) {
+                assert actualRuleCookies.size() == expectedNumberOfS42Rules
+            } else {
+                assert actualRuleCookies.isEmpty()
+            }
+        }
+        return originalProps.server42FlowRtt
+    }
+
+    static List<FlowEntry> s42SwitchRules(SwitchId swId) {
```

Comment:
What if we have methods `getServer42SwitchRules()` and `getExpectedServer42SwitchRules()` (this one would calculate expected rules based on switch properties) and compare them in tests?